templates/main/singleton/boil_queries.go.tpl

```diff
@@ -12,6 +12,13 @@ var dialect = drivers.Dialect{
 	UseCaseWhenExistsClause: {{.Dialect.UseCaseWhenExistsClause}},
 }
 
+{{if not .AutoColumns.Deleted}}
+    // This is a dummy variable to prevent unused regexp import error
+    var _ = &regexp.Regexp{}
+{{else}}
+    var removeSoftDeleteRgx = regexp.MustCompile("{{$.AutoColumns.Deleted}}[\"'`]? is null")
+{{- end}}
```

Comment:
This would be better alongside other global variables which are a little after the model definition.

Reply:
What do you exactly mean by `model`? The table's corresponding struct? Do you mean to have the variable replicated inside each table's generated file? Like here?

https://github.com/volatiletech/sqlboiler/blob/ca0d71ed51bab25289fb52dd217df8a4ffd859dc/templates/main/01_types.go.tpl#L45-L61

Would it make sense to set this globally in an `init()` function inside some singleton template? Create a new one for this use cases?

Reply:
> Would it make sense to set this globally in an `init()` function inside some singleton template? Create a new one for this use cases?

I mixed some stuff up. I think that would be the best solution for the current architecture, but this will break down if there are multiple generated models with different `deleted` columns, since they will all change the global `rgx` and whichever is last will win.

In a hypothetical `v5` the query mod to remove `deleted_at` should move into generated code, but that would need some changes to the query object.

Reply:
>  In a hypothetical v5 the query mod to remove deleted_at should move into generated code

I agree! Should I open a new issue labeled for v5 and so we can go over it later?

Reply:
Yes, that would be nice. Also if making any changes, send the PR to the `v5` branch.

Reply:
Gotcha! :+1: